network/gossip/libp2p/topology/topicBasedTopology_test.go

```diff
@@ -0,0 +1,248 @@
+		for _, v := range ids {
+			current = append(current, v.NodeID.String())
+		}
+		sort.Strings(current)
+
+		if previous == nil {
+			continue
+		}
+
+		// assert that a different seed generates a different topology
+		require.NotEqual(suite.T(), previous, current)
+	}
+}
+
+// TestConnectedness_NonClusterTopics checks whether graph components corresponding to a
+// non-cluster channel ID are individually connected.
+func (suite *TopicAwareTopologyTestSuite) TestConnectedness_NonClusterChannelID() {
+	channelID := engine.TestNetwork
+	// adjacency map keeps graph component of a single channel ID
+	channelIDAdjMap := make(map[flow.Identifier]flow.IdentityList)
+
+	for _, id := range suite.ids {
+		// creates a topic-based topology for node
+		top, err := topology.NewTopicBasedTopology(id.NodeID, suite.state)
+		require.NoError(suite.T(), err)
+
+		// samples subset of topology
+		subset, err := top.Subset(suite.ids, suite.fanout, channelID)
+		require.NoError(suite.T(), err)
+
+		channelIDAdjMap[id.NodeID] = subset
+	}
+
+	topology.CheckConnectednessByChannelID(suite.T(), channelIDAdjMap, suite.ids, channelID)
+}
+
+// TestConnectedness_NonClusterChannelID checks whether graph components corresponding to a
+// cluster channel ID are individually connected.
+func (suite *TopicAwareTopologyTestSuite) TestConnectedness_ClusterChannelID() {
+	// picks one cluster channel ID as sample
+	channelID := clusterChannelIDs(suite.T())[0]
+
+	// adjacency map keeps graph component of a single channel ID
+	channelIDAdjMap := make(map[flow.Identifier]flow.IdentityList)
+
+	// iterates over collection nodes
+	for _, id := range suite.ids.Filter(filter.HasRole(flow.RoleCollection)) {
+		// creates a channelID-based topology for node
+		top, err := topology.NewTopicBasedTopology(id.NodeID, suite.state)
+		require.NoError(suite.T(), err)
+
+		// samples subset of topology
+		subset, err := top.Subset(suite.ids, suite.fanout, channelID)
+		require.NoError(suite.T(), err)
+
+		channelIDAdjMap[id.NodeID] = subset
+	}
+
+	// check that each of the collection clusters forms a connected graph
+	for _, cluster := range suite.clusters {
+		suite.checkConnectednessByCluster(suite.T(), channelIDAdjMap, cluster)
+	}
+}
+
+// clusterChannelIDs is a test helper method that returns all cluster-based channel ids.
+func clusterChannelIDs(t *testing.T) []string {
+	ccids := make([]string, 0)
+	for _, channelID := range engine.ChannelIDs() {
+		if !engine.IsClusterChannelID(channelID) {
+			continue
+		}
+		ccids = append(ccids, channelID)
+	}
+
+	require.NotEmpty(t, ccids)
+	return ccids
+}
+
+// checkConnectednessByCluster is a test helper that checks all nodes belong to a cluster are connected.
+func (suite *TopicAwareTopologyTestSuite) checkConnectednessByCluster(t *testing.T,
+	adjMap map[flow.Identifier]flow.IdentityList,
+	cluster flow.IdentityList) {
+	topology.CheckGraphConnected(t,
+		adjMap,
+		suite.ids.Filter(filter.HasRole(flow.RoleCollection)),
+		filter.In(cluster))
```

Comment:
```suggestion
		suite.ids.Filter(filter.In(cluster))
```

All nodes in a cluster are anyway collection nodes